### ISSUE TITLE
Add optional dropout support

### ISSUE THREAD

**Issue by lecode-official** [lecode-official/pytorch-lottery-ticket-hypothesis#8]
In the original paper, the authors performed experiments with and without dropout. Right now, the models do not support dropout, so it should be able to optionally enable dropout during training.
